api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,34 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const currentToken = req.headers['x-access-token'];
+  if (!currentToken) {
+    throw new ClientError(401, 'authentication required');
+  }
+
+  const pulledPayload = jwt.verify(currentToken, process.env.TOKEN_SECRET);
+  if (!pulledPayload) {
+    throw new ClientError(401, 'invalid access token');
+  }
```

Comment:
You can leave this guard out, because `jwt.verify()` will throw its own error if the token is invalid.